### PR TITLE
feat: Async renderer

### PR DESCRIPTION
**Pull request by PuruVJ**

### CODE REVIEW

File: packages/site-kit/src/lib/markdown/renderer.js [hunk 1]
```diff
@@ -754,9 +765,21 @@ async function find_nearest_node_modules(start_path) {
 async function create_snippet_cache(should) {
 	const snippet_cache = (await find_nearest_node_modules(import.meta.url)) + '/.snippets';
 
-	try {
-		if (should) await mkdir(snippet_cache, { recursive: true });
-	} catch {}
+	// No local cache exists yet
+	if (!CACHE_MAP.size) {
+		try {
+			if (should) await mkdir(snippet_cache, { recursive: true });
```
Comment (dummdidumm) [inline packages/site-kit/src/lib/markdown/renderer.js:771]: The `should` should be on the outer `if(!CACHE_MAP.size)`

**Review by dummdidumm — COMMENTED** [review]
slight nit, else looks good

File: packages/site-kit/src/lib/markdown/renderer.js [hunk 2]
```diff
@@ -992,3 +1016,38 @@ function indent_multiline_comments(str) {
 		}
 	);
 }
+
+/**
+ *
```
Comment (benmccann) [inline packages/site-kit/src/lib/markdown/renderer.js:1021]:
```suggestion
```

File: packages/site-kit/src/lib/markdown/renderer.js [hunk 3]
```diff
@@ -992,3 +1016,38 @@ function indent_multiline_comments(str) {
...
+ * @param {string} inputString
+ * @param {RegExp} regex
+ * @param {(match: RegExpExecArray) => Promise<string>} asyncCallback
+ * @returns
```
Comment (benmccann) [inline packages/site-kit/src/lib/markdown/renderer.js:1025]:
```suggestion
```

File: packages/site-kit/src/lib/markdown/renderer.js (hunk 1 above)
Comment (PuruVJ) [reply packages/site-kit/src/lib/markdown/renderer.js]: Done

**Pull request merged**